Makefile

```diff
@@ -62,6 +62,10 @@ setup: dev_deps
 
 test:	test-unit test-integration
 
+# lint should happen after test-unit and test-examples as it relies on objects
+# being created during these phases
+test-all: test-unit test-race test-examples lint test-integration
```

Comment:
Makes sense, I will move this to `test` as it makes sense to run all tests there.